Makefile

```diff
@@ -67,6 +67,9 @@ format: venv
 format-check: venv
 	${PYTHON} -m black . --diff --check
 
+pre-commit: venv
+	pre-commit install --config .github/.pre-commit-config.yaml
```

Comment:
The goal when installing pre-commit hooks is to do it as early as possible (i.e. as soon as we can after installing the `pre-commit` with `pip install`), and to run the task whenever the `.github/.pre-commit-config.yaml` file is changed.

If you look at line 43, we do a check (when running `venv` target, which depends on `$(VENV_NAME)/bin/activate`) that `setup.py` has been changed. 

I believe the best option would be to add a dependency to `venv` on `.github/.pre-commit-config.yaml` which will re-run the full installation, as well as run `pre-commit install`

Reply:
This will also mean this make target can be deleted

Reply:
This has been moved up in to `venv`.